opal/mca/accelerator/accelerator.h

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (c) 2014-2021 Intel, Inc. All rights reserved.
+ * Copyright (c) 2015      Los Alamos National Security, LLC. All rights
+ *                         reserved.
+ * Copyright (c) 2017-2022 Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef OPAL_ACCELERATOR_H
+#define OPAL_ACCELERATOR_H
+
+#include "opal/class/opal_object.h"
+#include "opal/mca/mca.h"
+
+BEGIN_C_DECLS
+
+/**
+ * Accelerator flags
+ */
+/* Unified memory buffers */
+#define MCA_ACCELERATOR_FLAGS_UNIFIED_MEMORY 0x00000001
+
+/**
+ * Transfer types.
+ * UNSPEC - Not specified
+ * HTOH - Host to Host
+ * HTOD - Host to Device
+ * DTOH - Device to Host
+ * DTOD - Device to Device
+ */
+typedef enum {
+    MCA_ACCELERATOR_TRANSFER_UNSPEC = 0,
+    MCA_ACCELERATOR_TRANSFER_HTOH,
+    MCA_ACCELERATOR_TRANSFER_HTOD,
+    MCA_ACCELERATOR_TRANSFER_DTOH,
+    MCA_ACCELERATOR_TRANSFER_DTOD,
+} opal_accelerator_transfer_type_t;
+
+struct opal_accelerator_stream_t {
+    /* Stream object */
+    void *stream;
+};
+typedef struct opal_accelerator_stream_t opal_accelerator_stream_t;
+OBJ_CLASS_DECLARATION(opal_accelerator_stream_t);
+
+struct opal_accelerator_event_t {
+    /* Event object */
+    void *event;
+};
+typedef struct opal_accelerator_event_t opal_accelerator_event_t;
+OBJ_CLASS_DECLARATION(opal_accelerator_event_t);
+
+/**
+ * Check whether a pointer belongs to an accelerator or not.
+ * interfaces
+ *
+ * @param[IN] addr           Pointer to check
+ * @param[OUT] dev_id        Returns the device id against which the memory was allocated
+ * @param[OUT] flags         Set to 0 or an Accelerator flag to indicate additional
+ *                           information about the corresponding pointer
+ *
+ * @retval <0                An error has occurred.
+ * @retval 0                 The buffer does not belong to a managed buffer
+ *                           in device memory.
+ * @retval >0                The buffer belongs to a managed buffer in
+ *                           device memory.
+ */
+typedef int (*opal_accelerator_base_module_check_addr_fn_t)(
+    const void *addr, int *dev_id, uint64_t *flags);
+
+/**
+ * Creates a stream for asynchonous operations. This will also
+ * allocate the stream object.
+ *
+ * @param[IN] dev_id         Associated device for the stream
+ * @param[IN] stream         Stream to create
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ */
+typedef int (*opal_accelerator_base_module_create_stream_fn_t)(
```

Comment:
For habana, you also need to specify the stream type for different directions (h-d, d-h, d-d, h-h)

```
synStatus SYN_API_CALL synStreamCreate( synStreamHandle*       pStreamHandle,
                                        const synDeviceId      deviceId,
                                        const synStreamType    streamType,
                                        const uint32_t         flags );

```